core/src/main/scala/kafka/message/ByteBufferMessageSet.scala

```diff
@@ -78,7 +78,7 @@ object ByteBufferMessageSet {
   }
 
   /** Deep iterator that decompresses the message sets and adjusts timestamp and offset if needed. */
-  def deepIterator(wrapperMessageAndOffset: MessageAndOffset): Iterator[MessageAndOffset] = {
+  def deepIterator(wrapperMessageAndOffset: MessageAndOffset, ignorePartiallyConverted: Boolean = false): Iterator[MessageAndOffset] = {
```

Comment:
Downgrades are not supported, so it's OK not to handle that in my opinion. Maybe we should name it `ignoreMagicCorruption` or something like that to make it clear that this is not a good option to use. And then add a comment explaining the specifics. If you think the current name is better, I'm OK with that too, but we should add a comment to the scaladoc in any case.